streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamPeek.java

```diff
@@ -32,16 +33,16 @@ public KStreamPeek(final ForeachAction<K, V> action, final boolean forwardDownSt
     }
 
     @Override
-    public Processor<K, V> get() {
+    public Processor<K, V, K, V> get() {
         return new KStreamPeekProcessor();
     }
 
-    private class KStreamPeekProcessor extends AbstractProcessor<K, V> {
+    private class KStreamPeekProcessor extends AbstractProcessor<K, V, K, V> {
         @Override
-        public void process(final K key, final V value) {
-            action.apply(key, value);
+        public void process(final Record<K, V> record) {
+            action.apply(record.key(), record.value());
             if (forwardDownStream) {
-                context().forward(key, value);
+                context().forward(record);
             }
```

Comment:
Hmm. What do you think about splitting this up and creating a ForEachProcessor that implements `Processor<K, V, Void, Void>` and therefore cannot forward, then simplifying this class to _always_ forward?

Reply:
It makes sense. Will give a try.